### DISCUSSION TITLE
DynamicForm 1.07 released

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#3060] [Show and tell]
Version 1.0.7 - RELEASED
**New Feature**: you can now get your items reviewed.  Simple go to settings and turn the Start Rating on.  This will allow registered users to leave a review and a rating from 5 starts.  Clicking Read Reviews list all the reviews for that Item. From the settings you can specify aspect of the Review section.  There is now a new permission to allow reviews to be added.

![image](https://github.com/oqtane/oqtane.framework/assets/9447612/4053ba3f-8fd5-4058-b957-fcf1c4e2ba9c)

**Enhancement**: New Feature: when designing a Presenter it has always been a requirement to have data.  DynamicForm now has a Lipsum generator with lots of nice functions besides the standard word and paragraph.
**Enhancement**: To the DataList look and feel.  
**Enhancement**: To the Property Window look and feel.  
**Enhancement**: Emails out is now HTML formatted.  

**Bug**: Fixed DataCache in 1.0.6 not allowing module to run in WebAssembly. 
**Bug**: Fixed where Calculate component was not applying the CSSClass property to the value.
**Bug**: Fixed where Aggregator component was not calculating correctly for dynamic types.

https://www.studio-elf.net/dynamicform
https://www.studio-elf.net/dynamicform/forms
https://www.studio-elf.net/dynamicform/tips-and-tricks

**Comment by vnetonline** [comment]
@leigh-pointer I am unable to update your module on my https://amazingsolutions.com.au website, its stuck on 1.0.6.6 I can see the package being downloaded but it doesn't go through and install

here is the log 

```
[
  "\\StudioElf.DynamicForm.Client.Oqtane.dll",
  "\\StudioElf.DynamicForm.Client.Oqtane.pdb",
  "\\StudioElf.DynamicForm.Server.Oqtane.dll",
  "\\StudioElf.DynamicForm.Server.Oqtane.pdb",
  "\\StudioElf.DynamicForm.Shared.Oqtane.dll",
  "\\StudioElf.DynamicForm.Shared.Oqtane.pdb",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\Assets\\ImagePlacement.png",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\css\\Animations.css",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\css\\Colors.css",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\css\\Module.css",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\css\\Toolbox.css",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\css\\TreeView.css",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\js\\ConfigRequired.js",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\js\\jsInterop.js",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\js\\Module.js",
  "\\wwwroot\\Modules\\StudioElf.DynamicForm\\xslt\\DataTableToExcel.xslt",
  "\\wwwroot\\_content\\Placeholder.txt"
]
```

I restart the application and still on 1.0.6.6

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/ff7162a3-1a92-4ebd-8961-7716514b4dc1)

**Reply by leigh-pointer** [reply]
I think I had a similar issue.  So I restarted the web server. 
But if it has been unpacked, then Oqtane should pick it up.

**Reply by vnetonline** [reply]
Where are you hosting your site …. Azure ??

**Reply by leigh-pointer** [reply]
No I am hosting at InterServer.

**Reply by leigh-pointer** [reply]
@vnetonline So I have Oqtane with no DynamicForm module installed.  And installed from the Module Manager.  That went ok except it did show that it was version 1.0.0.  I then upgraded and now shows 1.0.7 and everything is working.

**Reply by vnetonline** [reply]
> No I am hosting at InterServer.

@leigh-pointer which plan are you using with interserve and are you doing CI/CD with your website ??

Thanks in advance for you help

**Reply by leigh-pointer** [reply]
I use Windows ASP.NET Web Hosting and not using CI/CD

**Comment by vnetonline** [comment]
Nope restarted the web server still says 1.0.6.6 what version did you make it work on .... I am running 4.0.1

**Reply by vnetonline** [reply]
@leigh-pointer & @sbwalker I think the issue is to do with #3058 for which i have submitted a pull request 4.0.1 has a bug should really release 4.0.2

uploading a module or theme doesn't work without this fix

**Reply by leigh-pointer** [reply]
Can you manually copy the package to the packages folder and restart, or extract the files and over-right them and restart?

**Comment by vnetonline** [comment]
Yup had to resort to doing that because upload doesn’t work in 4.0.1 with your module I deleted and installed again and I got 1.0.0 then I upgraded and it worked

**Reply by leigh-pointer** [reply]
Great. Let me know what you think!

**Comment by sbwalker** [comment]
@vnetonline the uplioad of packages was fixed in #3051 - it is not related to #3058

**Reply by vnetonline** [reply]
Yeah @sbwalker but #3051 didn't make it into 4.0.1 and I recently upgraded my site to 4.0.1 and the upload functionality is broken

**Reply by sbwalker** [reply]
#3051 will be in 4.0.2. In the meantime you can copy Nuget packages into the /Packages folder manually and they will be installed.

**Comment by sbwalker** [comment]
@leigh-pointer there is something strange going on with your site - when I try to browse using a link such as https://www.studio-elf.net/dynamicform the page is blank but when I navigate to https://www.studio-elf.net and choose the DynamicForm menu option it displays fine

**Reply by leigh-pointer** [reply]
@sbwalker I upgraded and that is fine, upgrade went well.  There is something going crazy with the SSL and I am trying to track it down.  
With Chorme getting this
```
Your connection is not private
Attackers might be trying to steal your information from www.studio-elf.net (for example, passwords, messages, or credit cards). [Learn more](chrome-error://chromewebdata/#)
NET::ERR_CERT_DATE_INVALID
```

Fire fox getting this
```
Did Not Connect: Potential Security Issue

Firefox detected an issue and did not continue to www.studio-elf.net. The website is either misconfigured or your computer clock is set to the wrong time.

It’s likely the website’s certificate is expired, which prevents Firefox from connecting securely.

What can you do about it?

www.studio-elf.net has a security policy called HTTP Strict Transport Security (HSTS), which means that Firefox can only connect to it securely. You can’t add an exception to visit this site.

The issue is most likely with the website, and there is nothing you can do to resolve it. You can notify the website’s administrator about the problem.

Learn more…
```

Intermitant with Edge.

Very annoying.  I thought it was a local issue so thank you for letting in know.

**Reply by leigh-pointer** [reply]
@sbwalker I am wondering if this could be a bug/issue.
https://www.sslshopper.com/ssl-checker.html?hostname=studio-elf.net is 100% fine.  So need to look at the framework.  Are there any other reported issues with SSL of access.

This is strange
![image](https://github.com/oqtane/oqtane.framework/assets/9447612/268c408e-d451-4939-a1ed-24f4e131edce)

![image](https://github.com/oqtane/oqtane.framework/assets/9447612/93706ec3-0c0c-46d4-9a52-dd0a8d0745e5)

**Comment by leigh-pointer** [comment]
@sbwalker found out what was causing that.  My meta tags where space padded on each line.  Remove the space padding now the page loads.

**Comment by sbwalker** [comment]
@leigh-pointer can you please explain what you mean by space padded - do you mean space delimited? Note that the framework parses the Head Content as it needs to remove any script references which may be included (as they need to be injected using JS Interop). Perhaps we need to add some better validation in the Site Settings for this field (ie. to ensure the content is well formatted HTML).

**Reply by vnetonline** [reply]
Yes better  validation and `Tirm()` would help

**Reply by leigh-pointer** [reply]
I had added the meta tags from a generator tool that I use. The tool padded each meta tag with 3 spaces. This paddling caused an error in the framework.

**Reply by sbwalker** [reply]
@leigh-pointer so you are saying that there were 3 spaces between each meta tag:

```
<meta name="description" content="Oqtane is a CMS and Application Framework for Blazor and .NET MAUI" />{space}{space}{space}<meta property="og:type" content="website" />
```

**Reply by leigh-pointer** [reply]
@sbwalker yes.  The Text I had paste into the Module Page Header had spaces.
![image](https://github.com/oqtane/oqtane.framework/assets/9447612/72cf4716-5d14-498e-a1d7-faee73a8b78d)
